MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -413,6 +421,9 @@ protected virtual void Dispose(bool disposing)
                     // Dispose of all remaining graphics resources before disposing of the graphics device
                     GraphicsResource.DisposeAll();
 
+                    // Clear the effect cache.
+                    EffectCache.Clear();

```

Comment:
Yep, I have a commit with exactly that fix sitting in a local branch already. I'll push it up as soon as this one is merged - just don't like to have too many PRs open at one time, it becomes difficult to manage.